dlt/destinations/utils.py

```diff
@@ -23,6 +23,12 @@ def ensure_resource(data: Any) -> DltResource:
     """Wraps `data` in a DltResource if it's not a DltResource already."""
     if isinstance(data, DltResource):
         return data
+    # prevent accidentally wrapping sources with adapters
+    if isinstance(data, DltSource):
+        raise Exception(
+            "You are trying to use an adapter on a dlt source. You can only use adapters on pure"
+            " data or dlt resources."
+        )
     resource_name = None if hasattr(data, "__name__") else "content"
```

Comment:
I think this else close should throw a warning. Otherwise, you get some unrecognizable table in the destination with the name `Content` out of nowhere :scream: